### PR TITLE
Replace multiple template/types/categories calls with single call to get all data and update redux store

### LINKED ISSUE #2472: [$60] Replace multiple template/types/categories calls with single call to get all data
For getting all the templates/types/categories data we call multiple public endpoints - 

- `/v4/projects/metadata/projectTemplates`
- `/v4/projects/metadata/productTemplates`
- `/v4/projects/metadata/projectTypes`
- `/v4/projects/metadata/productCategories`

As now we have support in our backend to replace all these calls with single call 

- `GET /v4/projects/metadata`

Ref - https://github.com/topcoder-platform/tc-project-service/issues/161

### PR DESCRIPTION
**Pull request by arun1595**
Fixes #2472

### CODE REVIEW

File: src/projects/create/containers/CreateContainer.jsx [hunk 1]
```diff
@@ -140,14 +139,9 @@ class CreateContainer extends React.Component {
...
-    if (templates.projectTemplates === null && !templates.isProjectTemplatesLoading) {
-      loadProjectTemplates()
-    }
-
-    // if categories are not loaded yet - then load them
-    if (templates.projectCategories === null && !templates.isProjectCategoriesLoading) {
-      loadProjectCategories()
+    // if metadata is not loaded yet - then load it
+    if (!templates.projectTemplates && !templates.isLoading) {
+      loadProjectsMetadata()
```
[the remark this answers is not in the document]
Comment (arun1595) [reply src/projects/create/containers/CreateContainer.jsx]: Derp :see_no_evil: 

Good catch btw. I've fixed the issue, PTAL :slightly_smiling_face:

File: src/reducers/templates.js [hunk 2]
```diff
@@ -1,40 +1,32 @@
...
-    return {...state,
-      isProjectCategoriesLoading: true,
-    }
-
-  case LOAD_PROJECT_CATEGORIES_SUCCESS:
-    return {...state,
-      projectCategories: action.payload,
-      isProjectCategoriesLoading: false,
+  case LOAD_PROJECTS_METADATA_SUCCESS: {
+    const { projectTemplates, productTemplates, projectTypes } = action.payload
```
Comment (vikasrohit) [inline src/reducers/templates.js]: @maxceem Don't we need product categories as well? I think we need that when rendering the modal for `Add Phase` action.

Comment (maxceem) [reply src/reducers/templates.js]: @vikasrohit so far we haven't used product categories in connect app anywhere. In particular `Add phase` modal is rendered as a flat list of products so also doesn't use product categories.

Comment (vikasrohit) [reply src/reducers/templates.js]: @maxceem I think we missed to merge the last CF challenge branch into dev where we actually implemented the functionality of rendering the products grouped by the categories. I think after that change need to incorporate the product categories as well. https://github.com/appirio-tech/connect-app/pull/2499

Comment (maxceem) [reply src/reducers/templates.js]: @vikasrohit Got it. So I'm going to accept this PR to let **arun1595** to work on the second part https://github.com/appirio-tech/connect-app/issues/2474 and I guess we will handle product categories in a separate issue after previous CF is merged to dev. Is it ok?

Comment (vikasrohit) [reply src/reducers/templates.js]: Yes, that is okay.

**Pull request merged**